src/event_object_movement.c

```diff
@@ -1234,12 +1232,12 @@ static u8 InitObjectEventStateFromTemplate(struct ObjectEventTemplate *template,
     objectEvent->range.as_nybbles.y = template->movementRangeY;
     objectEvent->trainerType = template->trainerType;
     objectEvent->trainerRange_berryTreeId = template->trainerRange_berryTreeId;
+#ifndef NONMATCHING
+    objectEvent->mapNum = mapNum; //Redundant, but needed to match
+#endif
```

Comment:
No need to wrap this statement in `#ifndef NONMATCHING`. I firmly believe that GameFreak would have done precisely this.